File: UNIDADE/flutter_application_finalProject/lib/iot/Cultivo.dart
class Cultivo {
  final int id;
  final String nome;
  final String tipo;
  final double temperaturaMinima;
  final double temperaturaMaxima;
  final double umidadeMinima;
  final double umidadeMaxima;
  final double umidadeSoloMinima;
  final double umidadeSoloMaxima;
  final bool habilitada;

  Cultivo({
    required this.id,
    required this.nome,
    required this.tipo,
    required this.temperaturaMinima,
    required this.temperaturaMaxima,
    required this.umidadeMinima,
    required this.umidadeMaxima,
    required this.umidadeSoloMinima,
    required this.umidadeSoloMaxima,
    required this.habilitada,
  });

  factory Cultivo.fromJson(Map<String, dynamic> json) {
    return Cultivo(
      id: json['id'],
      nome: json['nome'],
      tipo: json['tipo'],
      temperaturaMinima: json['temperaturaMinima'].toDouble(),
      temperaturaMaxima: json['temperaturaMaxima'].toDouble(),
      umidadeSoloMinima: (json['umidadeSoloMinima'] ?? 30.0).toDouble(),
      umidadeSoloMaxima: (json['umidadeSoloMaxima'] ?? 70.0).toDouble(),
      umidadeMinima: json['umidadeMinima'].toDouble(),
      umidadeMaxima: json['umidadeMaxima'].toDouble(),
      habilitada: json['habilitada'] == true || json['habilitada'] == 1,
    );
  }
}

File: UNIDADE/desktop_java_unidade/src/main/java/br/com/estufa/desktop/model/Cultivo.java
package br.com.estufa.desktop.model;

public class Cultivo {
    private Long id;
    private String nome;
    private String tipo;
    private float temperaturaMinima;
    private float temperaturaMaxima;
    private float umidadeMinima;
    private float umidadeMaxima;
    private float umidadeSoloMinima;
    private float umidadeSoloMaxima;
    private boolean habilitada;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public float getTemperaturaMinima() {
        return temperaturaMinima;
    }

    public void setTemperaturaMinima(float temperaturaMinima) {
        this.temperaturaMinima = temperaturaMinima;
    }

    public float getTemperaturaMaxima() {
        return temperaturaMaxima;
    }

    public void setTemperaturaMaxima(float temperaturaMaxima) {
        this.temperaturaMaxima = temperaturaMaxima;
    }

    public float getUmidadeMinima() {
        return umidadeMinima;
    }

    public void setUmidadeMinima(float umidadeMinima) {
        this.umidadeMinima = umidadeMinima;
    }

    public float getUmidadeMaxima() {
        return umidadeMaxima;
    }

    public void setUmidadeMaxima(float umidadeMaxima) {
        this.umidadeMaxima = umidadeMaxima;
    }

    public float getUmidadeSoloMinima() {
        return umidadeSoloMinima;
    }

    public void setUmidadeSoloMinima(float umidadeSoloMinima) {
        this.umidadeSoloMinima = umidadeSoloMinima;
    }

    public float getUmidadeSoloMaxima() {
        return umidadeSoloMaxima;
    }

    public void setUmidadeSoloMaxima(float umidadeSoloMaxima) {
        this.umidadeSoloMaxima = umidadeSoloMaxima;
    }

    public boolean isHabilitada() {
        return habilitada;
    }

    public void setHabilitada(boolean habilitada) {
        this.habilitada = habilitada;
    }
}
